pkg/cmd/workflow/view/view.go

```diff
@@ -0,0 +1,270 @@
+					return err
+				}
+			}
+			url = fmt.Sprintf("https://%s/%s/blob/%s/%s", hostname, ghrepo.FullName(repo), ref, workflow.Path)
+		} else {
+			baseName := filepath.Base(workflow.Path)
+			url = fmt.Sprintf("https://%s/%s/actions/workflows/%s", hostname, ghrepo.FullName(repo), baseName)
+		}
+		if opts.IO.IsStdoutTTY() {
+			fmt.Fprintf(opts.IO.Out, "Opening %s in your browser.\n", utils.DisplayURL(url))
+		}
+		return opts.Browser.Browse(url)
+	}
+
+	if opts.Yaml {
+		err = viewWorkflowContent(opts, client, workflow)
+	} else {
+		err = viewWorkflowInfo(opts, client, workflow)
+	}
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func viewWorkflowContent(opts *ViewOptions, client *api.Client, workflow *shared.Workflow) error {
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	opts.IO.StartProgressIndicator()
+	yaml, err := getWorkflowContent(client, repo, opts.Ref, workflow)
+	opts.IO.StopProgressIndicator()
+	if err != nil {
+		if s, ok := err.(api.HTTPError); ok {
+			if s.StatusCode == 404 {
+				base := filepath.Base(workflow.Path)
+				if opts.Ref != "" {
+					return fmt.Errorf("could not find workflow file %s on %s, try specifying a different ref", base, opts.Ref)
+				}
+				return fmt.Errorf("could not find workflow file %s, try specifying a branch or tag using --ref", base)
+			}
+		}
+		return fmt.Errorf("could not get workflow file content: %w", err)
+	}
+
+	theme := opts.IO.DetectTerminalTheme()
+	markdownStyle := markdown.GetStyle(theme)
+	if err := opts.IO.StartPager(); err != nil {
+		fmt.Fprintf(opts.IO.ErrOut, "starting pager failed: %v\n", err)
+	}
+	defer opts.IO.StopPager()
+
+	if !opts.Raw {
+		cs := opts.IO.ColorScheme()
+		out := opts.IO.Out
+
+		fileName := filepath.Base(workflow.Path)
+		fmt.Fprintf(out, "%s - %s\n", cs.Bold(workflow.Name), cs.Gray(fileName))
+		fmt.Fprintf(out, "ID: %s", cs.Cyanf("%d", workflow.ID))
+
+		codeBlock := fmt.Sprintf("```yaml\n%s\n```", yaml)
+		rendered, err := markdown.RenderWithOpts(codeBlock, markdownStyle,
+			markdown.RenderOpts{
+				markdown.WithoutIndentation(),
+				markdown.WithoutWrap(),
+			})
+		if err != nil {
+			return err
+		}
+		_, err = fmt.Fprint(opts.IO.Out, rendered)
+		return err
+	}
+
+	if _, err := fmt.Fprint(opts.IO.Out, yaml); err != nil {
+		return err
+	}
+
+	if !strings.HasSuffix(yaml, "\n") {
+		_, err := fmt.Fprint(opts.IO.Out, "\n")
+		return err
+	}
+
+	return nil
+}
+
+func viewWorkflowInfo(opts *ViewOptions, client *api.Client, workflow *shared.Workflow) error {
+	repo, err := opts.BaseRepo()
+	if err != nil {
+		return fmt.Errorf("could not determine base repo: %w", err)
+	}
+
+	opts.IO.StartProgressIndicator()
+	wr, err := getWorkflowRuns(client, repo, workflow)
+	opts.IO.StopProgressIndicator()
+	if err != nil {
+		return fmt.Errorf("failed to get runs: %w", err)
+	}
+
+	out := opts.IO.Out
+	cs := opts.IO.ColorScheme()
+	tp := utils.NewTablePrinter(opts.IO)
```

Comment:
We are only printing ~10 rows, should we start the pager?

Reply:
Additionally what should the notty output look like? Right now it is the same as the tty output.

Reply:
The pager might be nice for `--yaml` mode; I don't think it's needed otherwise.

For notty I'm very torn. I think for now you can just ignore the notty case and we'll see if someone complains.